pgxn/neon/control_plane_connector.c

```diff
@@ -670,14 +675,31 @@ HandleCreateRole(CreateRoleStmt *stmt)
 static void
 HandleAlterRole(AlterRoleStmt *stmt)
 {
-	const char *role_name = stmt->role->rolename;
+	char	   *role_name;
 	DefElem    *dpass;
 	ListCell   *option;
 	bool		found = false;
 	RoleEntry  *entry;
 
 	InitRoleTableIfNeeded();
 
+	switch (stmt->role->roletype)
+	{
+		case ROLESPEC_CSTRING:
+			role_name = pstrdup(stmt->role->rolename);
+			break;
+		case ROLESPEC_CURRENT_ROLE:
+		case ROLESPEC_CURRENT_USER:
+			role_name = GetUserNameFromId(GetUserId(), false);
+			break;
+		case ROLESPEC_SESSION_USER:
+			role_name = GetUserNameFromId(GetSessionUserId(), false);
+			break;
+		case ROLESPEC_PUBLIC:
+			role_name = pstrdup("public");
+			break;
+	}
```

Comment:
Do we need strings? Could we get away with Oids?

Reply:
We need the role name in the HTTP request to the control plane